### ISSUE TITLE
unparseable test files are silently skipped: check/run report success and exit 0

### ISSUE THREAD

**Issue by fzipi** [coreruleset/go-ftw#660]
## what

An unparseable test file is dropped with only a `WRN` line. Both `check` and `run` then report success and exit `0`, so a YAML syntax error can silently delete a rule's entire test coverage without failing CI.

`check` even prints `everything looks good!` for a run in which a file was dropped:

```
$ go-ftw check -d ./mixed
WRN Problem detected in file ./mixed/913100.yaml:
yaml: while scanning a quoted scalar at line 154, column 25: line 154, column 113: found unknown escape character

ftw/check: checked 1 files, everything looks good!
$ echo $?
0
```

`run` is worse, because the dropped file isn't even reflected in the skipped count:

```
$ go-ftw run -d ./mixed
🛠️ Starting tests!
🚀 Running go-ftw!
👉 executing tests in file
	running 911100-1: ✔ passed in 39.263167ms (RTT 56.612625ms)
	...
➕ run 8 total tests in 129.202959ms
⏭ skipped 0 tests
🎉 All tests successful!
$ echo $?
0
```

`./mixed` contains two files; only one ran. Nothing in that output says so.

## why it matters

This just happened for real on [coreruleset/coreruleset#4755](https://github.com/coreruleset/coreruleset/pull/4755). A `\/` escape inside a double-quoted YAML scalar made `tests/regression/tests/REQUEST-913-SCANNER-DETECTION/913100.yaml` unparseable. That's the only test file for rule 913100, so **all nine** of its tests stopped running — and both `regression (modsec2-apache)` and `regression (modsec3-nginx)` went green on the PR having tested nothing.

The PR was adding a new detection. Once the YAML was fixed, `913100-9` failed on `modsec2-apache` — a genuine engine-divergence bug that CI should have caught and didn't.

`check` gives no help either: `ftw/check: checked 321 files, everything looks good!` on the PR branch vs `322` on `main`. The only way to notice is to diff the file count between branches, which nobody does.

## root cause

[`test/files.go:38-43`](https://github.com/coreruleset/go-ftw/blob/main/test/files.go#L38-L43):

```go
ftwTest, err := GetTestFromYaml(yamlString, fileName)
if err != nil {
	log.Warn().Msgf("Problem detected in file %s:\n%v\n",
		filePath, err)
	continue
}
```

The error is swallowed and the loop continues. Only if *every* file fails does the function return an error (`no tests found`, line 48-50).

The function's own doc comment says the opposite of what it does:

```go
// GetTestsFromFiles will get the files to be processed.
// If some file has yaml error, will stop processing and
// return the error with the partial list of files read.
```

So this looks like an implementation/intent drift rather than a deliberate design choice.

## suggestion

Making a parse failure fatal would match the documented intent and is what I'd expect by default — a test file that can't be read is not a passing test file. If silently continuing needs to stay available for some workflow, an opt-in flag (`--ignore-unparseable` or similar) seems better than the current default.

At minimum, `check`/`run` shouldn't print `everything looks good!` / `All tests successful!` and exit `0` when files were dropped.

## reproduce

Put any file with a YAML syntax error next to a valid test file and run `go-ftw check -d <dir>` — e.g. a header value containing `\/` inside double quotes:

```yaml
User-Agent: "Mozilla/5.0 ... Chrome\/151.0.0.0 Safari\/537.36"
```

## version

`v2.5.0` (also reproduced on `v2.5.0-4-gee1e5c7`).

---

**ai disclosure**

- **tools used**: Claude (Opus 5)
- **assisted with**: reproducing the behaviour across `check`/`run`, capturing exit codes, drafting this issue
- **review performed**: reproduced with a two-file directory (one valid, one unparseable) against a live `modsec2-apache` stack and confirmed exit `0` from both `check` and `run`; confirmed the 321-vs-322 file-count delta on coreruleset/coreruleset#4755 vs `main`; read `test/files.go` to confirm the `continue` and the contradicting doc comment
